### ISSUE TITLE
issue with forecasting tsfit object ts_fit_forecast()

### ISSUE THREAD

**Issue by vpnagraj** [signaturescience/fiphde#123]
@stephenturner im stumped. was working on #121 and was rebuilding the sysdata when i hit a wall.

it happens at the vignette data lines:

```
library(tidyverse)
library(fiphde)

# vignette data
vd <- list()
# Get data include in vignette
vd$hosp <- get_hdgov_hosp(limitcols = TRUE)
# Limit to the previous epiweek
vd$hosp <- vd$hosp %>% filter(date <= "2022-05-28")

# Prep data
vd$prepped_hosp <-
  vd$hosp %>%
  prep_hdgov_hosp(statesonly=TRUE, min_per_week = 0, remove_incomplete = TRUE) %>%
  dplyr::filter(abbreviation != "DC")

# Make tsibble
vd$prepped_hosp_tsibble <- make_tsibble(vd$prepped_hosp,
                                        epiyear = epiyear,
                                        epiweek=epiweek,
                                        key=location)


# Fit model
vd$hosp_fitfor <- ts_fit_forecast(vd$prepped_hosp_tsibble,
                                  horizon=4L,
                                  outcome="flu.admits",
                                  covariates=c("hosp_rank", "ili_rank"))
```

```
Trimming to 2021-01-01
ARIMA  formula: flu.admits ~ PDQ(0, 0, 0) + pdq(1:2, 0:2, 0) + hosp_rank + ili_rank
ETS    formula: flu.admits ~ season(method = "N")
Error in UseMethod("forecast") : 
  no applicable method for 'forecast' applied to an object of class "c('tbl_df', 'tbl', 'data.frame')"
```

i stepped back through the code (not shown) and found that the model fitting all proceeds normally ... and then sure enough forecast doesnt like the object going into it:

https://github.com/signaturescience/fiphde/blob/main/R/forecast.R#L149

`tsfitfor` is a tibble (with individual models for each location in each row). 

so whats going on here? why havent we run into this before? 

FWIW im using fabletools 0.3.1. is this dependency version thing? waiting to resolve this => rebuild sysdata => recheck pkg before i merge the PR to close #120

**Comment by stephenturner** [comment]
Hrm, can you give me a sessionInfo()? I'm not hitting any problem with the forecast, but I _am_ hitting a problem with the next step...

```
> vd$ilidat <-
+   get_cdc_ili(region=c("state"), years=2019:2022) %>%
+   filter(region == "Hawaii") %>%
+   replace_ili_nowcast(., weeks_to_replace=1)
Error:
! object 'epiweek' not found
Run `rlang::last_error()` to see where the error occurred.
```


```
R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] fiphde_0.2.0  ggplot2_3.3.6 tidyr_1.2.0   readr_2.1.2   dplyr_1.0.9  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9           lattice_0.20-45      tsibble_1.1.1        lubridate_1.8.0     
 [5] here_1.0.1           prettyunits_1.1.1    class_7.3-20         digest_0.6.29       
 [9] assertthat_0.2.1     rprojroot_2.0.3      utf8_1.2.2           R6_2.5.1            
[13] e1071_1.7-11         httr_1.4.3           pillar_1.8.0         rlang_1.0.4         
[17] progress_1.2.2       curl_4.3.2           rstudioapi_0.13      labeling_0.4.2      
[21] MMWRweek_0.1.3       stringr_1.4.0        fabletools_0.3.2     bit_4.0.4           
[25] munsell_0.5.0        anytime_0.3.9        proxy_0.4-27         compiler_4.2.0      
[29] pkgconfig_2.0.3      urca_1.3-0           tidyselect_1.1.2     tibble_3.1.7        
[33] progressr_0.10.1     fansi_1.0.3          crayon_1.5.1         tzdb_0.3.0          
[37] withr_2.5.0          sf_1.0-7             cdcfluview_0.9.4     distributional_0.3.0
[41] grid_4.2.0           nlme_3.1-158         jsonlite_1.8.0       gtable_0.3.0        
[45] lifecycle_1.0.1      DBI_1.1.3            magrittr_2.0.3       units_0.8-0         
[49] scales_1.2.0         KernSmooth_2.23-20   fable_0.3.1          cli_3.3.0           
[53] stringi_1.7.8        vroom_1.5.7          farver_2.1.1         fs_1.5.2            
[57] xml2_1.3.3           ellipsis_0.3.2       generics_0.1.3       vctrs_0.4.1         
[61] tools_4.2.0          bit64_4.0.5          glue_1.6.2           purrr_0.3.4         
[65] hms_1.1.1            trending_0.0.3       parallel_4.2.0       colorspace_2.0-3    
[69] classInt_0.4-7       feasts_0.2.2         usethis_2.1.6       
```

**Comment by stephenturner** [comment]
The second problem there with the nowcast data is that there _is_ no nowcast data for the most recent two epiweeks.

https://delphi.cmu.edu/epidata/nowcast/?locations=nat&epiweeks=202235,202236

```
{ "epidata": [{"location":"nat","epiweek":202235,"value":1.75752,"std":0.209436},{"location":"nat","epiweek":202236,"value":1.81549,"std":0.207931}], "result": 1, "message": "success" }
```

But https://delphi.cmu.edu/epidata/nowcast/?locations=nat&epiweeks=202238,202239

```
{ "epidata": [], "result": -2, "message": "no results" }
```

Turns out the nowcast isn't very _now_: https://delphi.cmu.edu/nowcast/, only "forecasting" through sep 3 (today is Oct 4).

![image](https://user-images.githubusercontent.com/460076/193885239-e6899b86-07b2-4828-a522-84e57f47d980.png)

Keeping this issue open until we solve your original problem, but I say with the vignette data, we don't try rebuilding the sysdata, or else we remove the nowcasting bit and backdate it a bit to what's in the sysdata as of the previous release.

**Comment by stephenturner** [comment]
Pushed a change in c334f7d that simply filters the data to where the nowcasted ILI data was at the time the vignette was initially created, and no longer tries to nowcast. Probably a larger problem though if nowcasting is going to be over a month behind.

Still no issue with the forecasting problem, which was the original issue filed here. Let's look at your sessionInfo() and figure out what might be the problem.

**Comment by vpnagraj** [comment]
heres my `sessionInfo()`:

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: OS X  12.2.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] fiphde_0.2.0    forcats_0.5.0   stringr_1.4.0   dplyr_1.0.10    purrr_0.3.4     readr_2.0.2     tidyr_1.2.0     tibble_3.1.8   
 [9] ggplot2_3.3.6   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7           lattice_0.20-41      lubridate_1.8.0      tsibble_1.0.1        assertthat_0.2.1     digest_0.6.25       
 [7] utf8_1.1.4           R6_2.4.1             cellranger_1.1.0     backports_1.1.8      reprex_0.3.0         httr_1.4.2          
[13] pillar_1.7.0         rlang_1.0.5          curl_4.3             readxl_1.3.1         rstudioapi_0.13      blob_1.2.1          
[19] MMWRweek_0.1.3       fabletools_0.3.1     bit_4.0.4            munsell_0.5.0        broom_1.0.1          anytime_0.3.9       
[25] compiler_4.0.2       modelr_0.1.8         pkgconfig_2.0.3      urca_1.3-0           tidyselect_1.1.2     progressr_0.7.0     
[31] fansi_0.4.1          crayon_1.3.4         tzdb_0.1.2           dbplyr_1.4.4         withr_2.5.0          grid_4.0.2          
[37] distributional_0.2.2 nlme_3.1-148         jsonlite_1.7.2       gtable_0.3.0         lifecycle_1.0.1      DBI_1.1.0           
[43] magrittr_1.5         scales_1.1.1         cli_3.3.0            stringi_1.4.6        vroom_1.5.5          fable_0.3.0         
[49] farver_2.0.3         fs_1.5.0             xml2_1.3.2           ellipsis_0.3.2       generics_0.1.3       vctrs_0.4.1         
[55] tools_4.0.2          bit64_4.0.5          glue_1.6.2           hms_0.5.3            parallel_4.0.2       colorspace_1.4-1    
[61] rvest_0.3.5          haven_2.3.1          feasts_0.2.1   
```

**Comment by vpnagraj** [comment]
looks like youre running fabletools 0.3.2. im at 0.3.1. let me try incrementing and see if that fixes ...

**Comment by vpnagraj** [comment]
sheeesh. yeah that did it. after bumping to 0.3.2 i can run code i shared before. so i guess we need 0.3.2 ? should we add that as req in DESCRIPTION ??

what do you think @stephenturner

**Comment by stephenturner** [comment]
We're good again. Sysdata regenerated, vignette works again. @vpnagraj close this issue if/when you get your fabletools updated.

```
── R CMD check results ──────────────────────────────────────────────────── fiphde 0.2.0 ────
Duration: 34.1s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔

R CMD check succeeded
```

**Comment by stephenturner** [comment]
Yeah fixed in e746dd8 , R CMD CHECK OK
